### ISSUE TITLE
Implement basic tests

### ISSUE THREAD

**Issue by csadorf** [aiidalab/aiidalab-widgets-base#347]
- [X] Implement basic tests using `pytest-docker` for existing notebooks, replacing the `aiidalab-test-app-action`.
- [ ] document how to implement tests as part of the aiidalab docs.
- [x] For this repository we should focus on unit testing individual widgets by manipulating their traits and checking the expected widget state. This will ensure API consistency and widget behavioral correctness.
- [x] #440

**Comment by csadorf** [comment]
For this repository we should focus on unit testing individual widgets by manipulating their traits and checking the expected widget state. This will ensure API consistency and widget behavioral correctness.

Testing the interface via selenium or similar should only be necessary in specific cases.

**Comment by csadorf** [comment]
For this library, we should also measure the coverage, similar to aiidalab-launch and eventually aim for a test coverage of at least 75-80%. I would consider this issue completed if we reach a coverage of 15%.

**Comment by unkcpz** [comment]
As a part of https://github.com/aiidalab/aiidalab-widgets-base/pull/344, I implement the tests by replacing `aiidalab-test-app-action` with `pytest-selenium` + `pytest-docker`. Seems all work well.

**Comment by yakutovicha** [comment]
since #344 is merged, should we close this @unkcpz?

**Comment by unkcpz** [comment]
@danielhollas thanks for pointing this out. I think you are right, no need for another issue, I'll reopen this one and change the title and use the checkboxes from your comment if you don't mind.

**Comment by danielhollas** [comment]
Blocked on #318

**Comment by unkcpz** [comment]
@yakutovicha From the checkboxes above, what we missing is the docs. I personally don't think it is useful to document how to write a unit test, it is all in the codes.

**Comment by unkcpz** [comment]
I would consider this as completed.
